tests/test_datetime_fields.py

```diff
@@ -20,8 +20,14 @@ def tearDown(self):
 
     def test_ad_hoc_model(self):
         self.database.insert([
-            ModelWithDate(date_field='2016-08-30', datetime_field='2016-08-30 03:50:00'),
-            ModelWithDate(date_field='2016-08-31', datetime_field='2016-08-31 01:30:00')
+            ModelWithDate(
+                date_field='2016-08-30',
+                datetime_field='2016-08-30 03:50:00',
+                datetime64_field='2016-08-30 03:50:00.001'),
+            ModelWithDate(
+                date_field='2016-08-31',
+                datetime_field='2016-08-31 01:30:00',
+                datetime64_field='2016-08-31 01:30:00.002')
```

Comment:
I think we need more tests with different precision values, and also with timezones. Timezones always present a huge potential for tricky bugs.